src/components/BookForm.js

```diff
@@ -1,53 +1,67 @@
-import React from 'react';
-import { useDispatch, useSelector } from 'react-redux';
+import React, { useState } from 'react';
 import { v4 as uuidv4 } from 'uuid';
-import { addBook } from '../redux/books/books';
+import { useDispatch } from 'react-redux';
+import { addBookToAPI } from '../redux/books/books';
 
-const categories = ['Action', 'Science-fiction', 'Business', 'Romance'];
+const categories = ['Action', 'Science Fiction', 'Economy'];
 
 const BookForm = () => {
-  const books = useSelector((state) => state.booksReducer);
   const dispatch = useDispatch();
+  const [title, setTitle] = useState('');
+  const [author, setAuthor] = useState('');
+  const [category, setCategory] = useState('');
+
   const submitBookToStore = (e) => {
     e.preventDefault();
-    if (e.target.parentNode.lastChild[0].value && e.target.parentNode.lastChild[1].value
-      && e.target.parentNode.lastChild[2].value) {
+    if (title && category && author) {
       const newBook = {
-        id: uuidv4(),
-        title: e.target.parentNode.lastChild[0].value,
-        author: e.target.parentNode.lastChild[1].value,
-        category: e.target.parentNode.lastChild[2].value,
+        item_id: uuidv4(),
+        title: {
+          title,
+          author,
+        },
+        category,
       };
-      dispatch(addBook(newBook));
-      e.target.parentNode.lastChild[0].value = '';
-      e.target.parentNode.lastChild[1].value = '';
+      dispatch(addBookToAPI(newBook));
+      setTitle('');
+      setAuthor('');
+      setCategory('');
     }
   };
-
   return (
-    <>
-      <h3 className="add-new">Add newbook</h3>
-      <form className="form-container" onSubmit={submitBookToStore}>
+    <div className="create-book">
+      <h2 className="add-book">Add new Book</h2>
+      <form
+        className="form"
+        onSubmit={submitBookToStore}
+      >
         <input
+          className="input"
           type="text"
           placeholder="Book title"
           name="title"
-          value={books.title}
+          value={title}
+          onChange={(e) => setTitle(e.target.value)}
```

Comment:
### OPTIONAL

- [ ] You use the onChange function in a few inputs. 
You can create one function inside the component and you can use it anywhere you want.

Example:

`  const handleChange = (event) => {
    const { name, value } = event.target;
    setState({ ...state, [name]: value });
  };`

Reply:
Thank you for this, I will take that into consideration